File: tests/fixtures/lang_samples/typescript/classes.ts
// Fixtures for class-level (LCOM4) walker tests.

class Cohesive {
  total = 0;
  count = 0;

  add(n: number): void {
    this.total += n;
    this.count += 1;
  }

  average(): number {
    return this.count ? this.total / this.count : 0;
  }

  reset(): void {
    this.total = 0;
    this.count = 0;
  }

  describe(): string {
    return `${this.count} items, avg ${this.average()}`;
  }
}

// Fields are declared (not assigned in a constructor) so no single
// method bridges the two clusters → LCOM4 == 3.
class Splintered {
  a = 0;
  b = 0;

  setA(v: number): void {
    this.a = v;
  }

  getA(): number {
    return this.a;
  }

  setB(v: number): void {
    this.b = v;
  }

  getB(): number {
    return this.b;
  }

  loner(): number {
    return 42;
  }
}

File: tests/fixtures/lang_samples/python/classes.py
"""Fixtures for class-level (LCOM4 / god-class) walker tests."""


class Cohesive:
    """All methods collaborate around shared state → LCOM4 == 1."""

    def __init__(self):
        self.total = 0
        self.count = 0

    def add(self, n):
        self.total += n
        self.count += 1

    def average(self):
        return self.total / self.count if self.count else 0

    def reset(self):
        self.total = 0
        self.count = 0

    def describe(self):
        return f"{self.count} items, avg {self.average()}"


class Splintered:
    """Two disjoint method clusters plus a loner → LCOM4 == 3.

    {set_a, get_a} share self.a; {set_b, get_b} share self.b; loner
    touches neither. Fields are class-level so no constructor bridges the
    clusters.
    """

    a = 0
    b = 0

    def set_a(self, v):
        self.a = v

    def get_a(self):
        return self.a

    def set_b(self, v):
        self.b = v

    def get_b(self):
        return self.b

    def loner(self):
        return 42
